devenv/local/docker-compose/tests/devnet-liveness.sh

```diff
@@ -0,0 +1,261 @@
+
+# echo "\nGET STACKS NODE INFO:"
+# echo $GET_STACKS_NODE_INFO | jq 'del(.stackerdbs)'
+# echo "\t\t.\n\t\t.\n  \033[1;32m<<\033[0m \033[1;35mLong Output Supressed\033[0m \033[1;32m>>\033[0m \n\t\t.\n\t\t."
+
+# STX_SYNC_WITH_BTC_UTXO_SUCCESS=$(echo $GET_STACKS_NODE_INFO | jq -r '.stacks_tip_height != 0')
+# STX_SYNC_WITH_BTC_UTXO_SUCCESS_FRMT=$([ "$STX_SYNC_WITH_BTC_UTXO_SUCCESS" == "true" ] && echo "\033[1;32m$STX_SYNC_WITH_BTC_UTXO_SUCCESS\033[0m ✅" || echo "\033[1;31m$STX_SYNC_WITH_BTC_UTXO_SUCCESS\033[0m❌") 
+
+# echo "\033[1mSTX_SYNC_WITH_BTC_UTXO_SUCCESS\033[0m: $STX_SYNC_WITH_BTC_UTXO_SUCCESS_FRMT"
+# echo "\n"
+
+
+## (LOGS APPROACH)
+STACKS_DOCKER_LOGS=$(docker logs stacks 2>/dev/null)
+
+
+STX_SYNC_WITH_BTC_UTXO_SUCCESS=false
+STX_SYNC_WITH_BTC_UTXO_SUCCESS_FRMT=$(echo "\033[1;31m$STX_SYNC_WITH_BTC_UTXO_SUCCESS\033[0m❌")
+if [[ $STACKS_DOCKER_LOGS == *"UTXOs found"* ]]; then
+    STX_SYNC_WITH_BTC_UTXO_SUCCESS=true
+    echo "Stacks || UTXOs found - will run as a Miner node"
+    STX_SYNC_WITH_BTC_UTXO_SUCCESS_FRMT=$(echo "\033[1;32m$STX_SYNC_WITH_BTC_UTXO_SUCCESS\033[0m ✅")
+fi
+
+echo "\033[1mSTX_SYNC_WITH_BTC_UTXO_SUCCESS\033[0m: $STX_SYNC_WITH_BTC_UTXO_SUCCESS_FRMT"
+echo "\n"
+
+
+
+
+echo " ---------------------------------------------------------------"
+echo "| => (8) 🔬 TEST: [CHECK STACKS-API LIVENESS]  |"
+echo " ---------------------------------------------------------------"
+
+
+GET_STACKS_API_PING=$(curl -s "http://localhost:3700")
+
+
+echo "\nGET STACKS API PING:"
+echo $GET_STACKS_API_PING | jq
+
+
+STACKS_API_LIVENESS_SUCCESS=$(echo $GET_STACKS_API_PING | jq -r '.status == "ready"')
+STACKS_API_LIVENESS_SUCCESS_FRMT=$([ "$STACKS_API_LIVENESS_SUCCESS" == "true" ] && echo "\033[1;32m$STACKS_API_LIVENESS_SUCCESS\033[0m ✅" || echo "\033[1;31m$STACKS_API_LIVENESS_SUCCESS\033[0m❌") 
+
+
+echo "\033[1mSTACKS_API_LIVENESS_SUCCESS\033[0m: $STACKS_API_LIVENESS_SUCCESS_FRMT"
+echo "\n"
+
+
+
+
+
+
+
+
+
+echo " -----------------------------------------------------------------"
+echo "| => (9) 🔬 TEST: [CHECK IF STACKS-API IS CONNECTED TO POSTGRES]  |"
+echo " -----------------------------------------------------------------"
+
+
+STACKS_API_DOCKER_LOGS=$(docker logs stacks-api 2>/dev/null)
+
+
+STACKS_API_CONNECTED_TO_PG_SUCCESS=false
+STACKS_API_CONNECTED_TO_PG_SUCCESS_FRMT=$(echo "\033[1;31m$STACKS_API_CONNECTED_TO_PG_SUCCESS\033[0m❌")
+if [[ $STACKS_API_DOCKER_LOGS == *"PgNotifier connected"* ]]; then
+    STACKS_API_CONNECTED_TO_PG_SUCCESS=true
+    echo "Stacks-API || PgNotifier connected"
+    STACKS_API_CONNECTED_TO_PG_SUCCESS_FRMT=$(echo "\033[1;32m$STACKS_API_CONNECTED_TO_PG_SUCCESS\033[0m ✅")
+fi
+
+
+echo "\033[1mSTACKS_API_CONNECTED_TO_PG_SUCCESS\033[0m: $STACKS_API_CONNECTED_TO_PG_SUCCESS_FRMT"
+echo "\n"
+
+
+echo "----------------------------------------------------------"
+echo "|                        SUMMARY                         |"
+echo "----------------------------------------------------------"
```

Comment:
Because this is a test file, it should exit with code 1 unless every test passes

Reply:
Tests return a 0 or 1 now